### ISSUE TITLE
Point does not implement == operator

### ISSUE THREAD

**Issue by marcojakob** [bp74/StageXL#177]
The `stagexl.geom.point.Point` class implements the `math.Point` interface. The StageXL Point is missing the equals operator method. This leads to unexpected behavior because when two StageXL Points are compared the `Object`s equality method is called which checks for `identical`. 

To fix this the following method should be implemented in StageXL Point: 

```dart
  bool operator ==(other) {
    if (other is !Point) return false;
    return x == other.x && y == other.y;
  }
```

There is also an `equals(...)` method in the StageXL Point which doesn't really make sense to me. Is this for compatibility reasons?

There might be other classes having the same issue, like `Rectangle`,...

**Comment by bp74** [comment]
Yes you are right, we should add the operator. I will take a look at it over the next few days.

**Comment by marcojakob** [comment]
Is the `equals(...)` method for compatibility reasons to AS3? Because this is not really a Dart-y way for comparing. If it isn't really necessary for compatibility, we could mark it as `@deprecated` and point the users to the `==` operator.

**Comment by bp74** [comment]
Yes, the equals methods (and some others) were there in the beginning for full Flash compatibility - this was before the introduction of Point and Rectangle in the dart:math library. Later we did some changes to make the StageXL Point and Rectangle compatible to the Dart Point and Rectangle types.

The equals method is from Flash.
http://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/geom/Point.html#equals()

We already deprecated some of the other methods that are available in Flash but are named differently in Dart. Therefore the equals method would be a good candidate to deprecate too.

**Comment by bp74** [comment]
Okay it's done. 

Point.add, Point.subtract, Point.equals are deprecated, 
Point.operator +, -, *, == were added.

Rectangle.equals is deprecated.
Rectangle.operator == was added.

**Comment by marcojakob** [comment]
Nice, thank you.

**Comment by bp74** [comment]
We shipped this with StageXL 0.11.0

**Comment by xavierhainaux** [comment]
Just for info Bernhard,
With this change, the dart2js compiler adds a log in the console.
`Hint: 2 hint(s) suppressed in package:stagexl.`

If I compile with `--show-package-warnings`:
```
packages/stagexl/src/geom/rectangle.dart:29:17:
Hint: The class 'Rectangle' overrides 'operator==', but not 'get hashCode'.
  bool operator ==(other) {
                ^^
packages/stagexl/src/geom/point.dart:37:17:
Hint: The class 'Point' overrides 'operator==', but not 'get hashCode'.
  bool operator ==(other) {
```

Nothing important I guess :-)

**Comment by bp74** [comment]
Okay this is very strange. I use the analyzer hints all the time but i didn't saw it. I will take a look and fix it asap!

**Comment by bp74** [comment]
Do you use Dart Editor or Webstorm?

**Comment by xavierhainaux** [comment]
This is not with the analyser actually, this is a warning with dart2js in command line.
like:
 `dart2js --show-package-warnings -o my_game.js my_game.dart`

**Comment by bp74** [comment]
Right, now i see it too :) I never looked at the dart2js warnings, just the analyzer warnings and hints. I will fix it and probably open an issue on dartbug.com - the analyzer should show those hints too.

**Comment by bp74** [comment]
Reported an issue to get the same warning with the analyzer:
https://code.google.com/p/dart/issues/detail?id=22998

**Comment by bp74** [comment]
Okay should be fixed! The new version is 0.11.0+1 and it is available on Pub.
Thanks for the bug report!
